src/main/java/wolox/reactortraining/services/BotService.java

```diff
@@ -38,4 +66,76 @@
             .bodyToMono(BotResponse.class);
     }
 
+    @Override
+    public Mono<Void> createBot(BotCreationDto botCreationDto) {
+        Mono<User> userMono = createUser(botCreationDto.getUsername());
+        Mono<Void> createBot = createBot(botCreationDto.getBotName(), botCreationDto.getTopics());
+
+        return userMono
+            .flatMap(user ->
+                Mono.zip(createTopics(user, botCreationDto.getTopics()), Mono.just(user)))
+            .flatMap(tuple -> {
+                User user = tuple.getT2();
+                List<Topic> topics = tuple.getT1();
+
+                user.addTopics(topics);
+
+                return userRepository.save(user);
+            })
+            .then(createBot);
+    }
+
+    private Mono<User> createUser(String username) {
+        UserDto userDto = new UserDto();
+        userDto.setUsername(username);
+
+        User user = new User(userDto);
+
+        return userRepository.insert(user);
+    }
+
+    private Mono<List<Topic>> createTopics(User user, List<String> topics) {
+        return Flux
+            .fromIterable(topics)
+            .map(description -> {
+                TopicDto topicDto = new TopicDto();
+                topicDto.setDescription(description);
+                return topicDto;
+            })
+            .map(dto -> {
+                Topic topic = new Topic(dto);
+                topic.addUser(user);
+                return topic;
+            })
+            .flatMap(topicRepository::insert)
+            .collectList();
+    }
+
+    private Mono<Void> createBot(String botName, List<String> topics) {
+        return twitterService
+            .getTweetsStreamPipe()
+            .map(Tweet::getText)
+            .take(500)
+            .filter(text -> {
+                for (String topic : topics) {
```

Comment:
You should do a refactor on this, another method with descriptive name for claritier